### PR TITLE
fix(ATCNet): correct TCN in/out channels, final-layer in_features, and residual skip conv

### PR DESCRIPTION
**Pull request by Altaheri**
### Summary
This PR fixes several issues in `braindecode/models/atcnet.py` that could lead to channel/feature mismatches and an incorrect residual connection:

1) **TCN residual stack input channels**
   - First `_TCNResidualBlock` should receive `in_channels=F2`, subsequent blocks `in_channels=tcn_n_filters`.
   - **Fix:** `in_channels=self.F2 if i == 0 else tcn_n_filters`

2) **Final classifier `in_features`**
   - Classifier should consume `tcn_n_filters` (× `n_windows` in the multi-window case), not `F2`.
   - **Fix:** `in_features=self.tcn_n_filters * self.n_windows` (sequence) and `in_features=self.tcn_n_filters` (single-window)

3) **Residual path 1×1 conv & application**
   - The 1×1 conv must take `in_channels=<skip tensor channels>` and output `n_filters`.
   - It must be applied to the **skip** tensor `X`, not to `out`.
   - **Fixes:**
     - `self.reshaping_conv = nn.Conv1d(in_channels=in_channels, out_channels=n_filters, kernel_size=1, padding="same")`
     - In `forward`: `X = self.reshaping_conv(X)` (not `out = self.reshaping_conv(out)`)

### Diff
- 6 additions, 5 deletions (11 lines total) in `braindecode/models/atcnet.py`, matching the changes above.

### Rationale
- Ensures channel consistency across TCN residual blocks.
- Aligns final classifier features with actual TCN output width.
- Corrects residual semantics (reshape the skip path when needed).

### Notes
- No API changes; self-contained bugfix.
- Thank you for maintaining Braindecode and for the PyTorch ATCNet implementation.

### CODE REVIEW

File: braindecode/models/atcnet.py [hunk 1]
```diff
@@ -370,11 +370,11 @@ def __init__(
...
-                            in_channels=self.F2,
-                            kernel_size=self.tcn_kernel_size,
-                            n_filters=self.tcn_n_filters,
-                            dropout=self.tcn_dropout,
-                            activation=self.tcn_activation,
+                            in_channels=self.F2 if i == 0 else tcn_n_filters,
+                            kernel_size=tcn_kernel_size,
+                            n_filters=tcn_n_filters,
+                            dropout=tcn_drop_prob,
+                            activation=tcn_activation,
```
Comment (bruAristimunha) [inline braindecode/models/atcnet.py]:
```suggestion
                            in_channels=self.F2 if i == 0 else self.tcn_n_filters,
                            kernel_size=self.tcn_kernel_size,
                            n_filters=self.tcn_n_filters,
                            dropout=self.tcn_drop_prob,
                            activation=self.tcn_activation,
```

**Comment by bruAristimunha** [comment]
hey @Altaheri,

Thank you for pushing this. I noticed some inconsistencies with Atcnet when we perform some experiments, but we never manage to understand the reason. 

Can you please fill in the whats new file to record the change?

**Comment by bruAristimunha** [comment]
@josemaurici FYI

File: braindecode/models/atcnet.py [hunk 2]
```diff
@@ -370,10 +370,10 @@ def __init__(
                 nn.Sequential(
                     *[
                         _TCNResidualBlock(
-                            in_channels=self.F2,
+                            in_channels=self.F2 if i == 0 else self.tcn_n_filters,
                             kernel_size=self.tcn_kernel_size,
                             n_filters=self.tcn_n_filters,
-                            dropout=self.tcn_dropout,
+                            dropout=self.tcn_drop_prob,
```
Comment (bruAristimunha) [inline braindecode/models/atcnet.py]:
```suggestion
                            dropout=self.tcn_dropout,
```

**Comment by Altaheri** [comment]
> hey @Altaheri,
> 
> Thank you for pushing this. I noticed some inconsistencies with Atcnet when we perform some experiments, but we never manage to understand the reason.
> 
> Can you please fill in the whats new file to record the change?

Hi @bruAristimunha,

Thanks for the quick review! 

I haven’t yet gone through the entire implementation in detail, but the changes I did here address logical bugs that would trigger an error whenever the input dimension to the TCN differs from the number of TCN filters. In the current code, this works only if the input and TCN dimensions happen to match, but fails otherwise.

I’ll also cross-check this implementation against the version in [TCFormer](https://github.com/Altaheri/TCFormer/blob/main/models/atcnet.py), which has shown consistent performance with the original TensorFlow code. After verifying, I’ll update you if further adjustments are required.

For now, this fix specifically resolves the input dimension mismatch in the TCN.

**Comment by bruAristimunha** [comment]
We would be happy if you could contribute to the new model to braindecode as well.

**Comment by Altaheri** [comment]
That would be great! I’d be happy to contribute the new model to Braindecode.

Thanks again for reviewing and for maintaining this excellent project!

**Comment by bruAristimunha** [comment]
Perfect, contribution made!

I hope to see you again soon!

Please use the version of braindecode available in the source code:

```bash
pip install git+https://github.com/braindecode/braindecode
```

or use the pre-release version in 2 hours:

```bash
pip install --pre braindecode
```
